### PR TITLE
Use SetBucketVersioning API style

### PR DESCRIPTION
**Pull request by vadmeste**
This refactor removes EnableVersioning() and DisableVersioning() and uses the following style instead:

```go
if err := s3Client.SetBucketVersioning(ctx, bucketName, minio.VersioningEnabled) ; err != nil {
...
}

if err := s3Client.SetBucketVersioning(ctx, bucketName, minio.VersioningEnabled, minio.VersioningMFA); err != nil {
...
}
```

### CODE REVIEW

File: api-put-bucket.go [hunk 1]
```diff
@@ -468,12 +468,27 @@ func (c Client) setVersioning(ctx context.Context, bucketName string, config []b
...
 }
 
-// EnableVersioning - Enable object versioning in given bucket.
-func (c Client) EnableVersioning(ctx context.Context, bucketName string) error {
-	return c.setVersioning(ctx, bucketName, versionEnableConfig, versionEnableConfigLen, versionEnableConfigMD5Sum, versionEnableConfigSHA256)
-}
+type VersioningStatus int
+
+const (
+	VersioningEnabled = iota
```
Comment (klauspost) [inline api-put-bucket.go:474]: Maybe swap them so VersioningSuspended is 0.

File: api-put-bucket.go [hunk 2]
```diff
@@ -468,12 +468,27 @@ func (c Client) setVersioning(ctx context.Context, bucketName string, config []b
...
+	VersioningSuspended
+)
+
+type VersioningOpts int
+
+const (
+	VersioningMFA = iota
```
[the remark this answers is not in the document]
Comment (klauspost) [reply api-put-bucket.go:481]: What is this?

**Comment by vadmeste** [comment]
@klauspost I am not sure if we will take this proposal in the first place, let's review it when we are sure that this API change is approved.

**Comment by harshavardhana** [comment]
@vadmeste PR #1338 supersedes this approach.

**Comment by vadmeste** [comment]
> @vadmeste PR #1338 supersedes this approach.

Closing then.

**Pull request closed without merging**